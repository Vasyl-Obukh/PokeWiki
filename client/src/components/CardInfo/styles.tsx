import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Info = styled.div`
  padding: 10px 10px 0;
  color: #333;
`;

export const Name = styled(Link)`
  display: block;
  margin: 0;
  font-size: 18px;
  color: #333;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  transition: all .2s;
  
  &:hover {
    color: #777;
  }
`;

export const Entry = styled.div`
  margin: 5px 0;
  text-align: center;
  color: #444;
`;

export const Key = styled.div`
  font-weight: bold;
  &::after {
    content: ':';
  }
`;

export const Values = styled.ul`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding: 0;
  margin: 0;
`;

export const Value = styled.li`
  list-style: none;
  padding: 5px 10px;
  font-size: 14px;
  
  &:not(:last-child) {
    border-right: 2px solid lightgray;
  }
`;