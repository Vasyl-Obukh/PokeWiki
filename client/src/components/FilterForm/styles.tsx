import styled from 'styled-components';
import {FormSection} from 'redux-form';

export const Form = styled.form`
  padding: 0 50px;
`;

export const StyledFormSection = styled(FormSection)`
  display: grid;
  grid-template-columns: 1fr 1fr;
`;

export const Label = styled.label`
  margin-left: 5px;
`;

export const FormButton = styled.button`
  padding: 5px 10px;
  margin-top: 25px;
  width: 100%;
  border-radius: 20px;
  background-color: ${props => props.theme.backgroundPrimary};
  text-transform: capitalize;
  color: white;
  border: none;
  cursor: pointer;
`;