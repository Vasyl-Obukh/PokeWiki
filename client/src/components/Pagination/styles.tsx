import styled from 'styled-components';

export const List = styled.ul`
  display: flex;
  padding: 0;
  justify-content: center;
`;

export const ListItem = styled.li`
  list-style: none;
  color: white;
`;

export const Item = styled.span<{current?: boolean, to?: string, as?: object}>`
  display: flex;
  width: 30px;
  height: 30px;
  border-radius: 50%;
  margin: 0 5px;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.current ? 'white' : props.theme.backgroundPrimary};
  border: ${props => props.current ? `1px solid ${props.theme.backgroundPrimary}` : 'none' };
  color: ${props => props.current ? props.theme.backgroundPrimary : 'inherit'};
  text-decoration: none;
`;