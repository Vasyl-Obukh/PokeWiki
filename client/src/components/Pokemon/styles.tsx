import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  padding-bottom: 50px;
  flex-direction: column;
  align-items: center;
`;

export const Image = styled.img`
  width: 275px;
  height: 275px;
  border-radius: 50%;
  background-color: #f0f0f0;
  box-shadow: 0 0 5px darkslategray;
`;

export const Info = styled.div`
  display: grid;
  margin-top: 25px;
  grid-template-columns: auto auto;
  grid-column-gap: 5px;
  grid-row-gap: 10px;
  justify-content: center;
`;

export const InfoKey = styled.div`
  font-weight: bold;
  text-transform: capitalize;
  text-align: right;
  
  &::after {
    content: ':';
  }
`;

export const InfoValue = styled.div`
  font-style: italic;
  font-weight: ${props => props.primary ? 'bold' : 'normal'};
  text-transform: ${props => props.primary ? 'uppercase' : 'none'};
`;

export const ValueItem = styled.span`
  padding: 5px;
  
  &:not(:first-child) {
    border-left: 1px solid gray;
  }
`;