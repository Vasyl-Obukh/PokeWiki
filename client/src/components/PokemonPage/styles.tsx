import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: 50px 15px;
  height: 100%;
`;

export const Main = styled.main`
  padding: 25px 10px;
  border-radius: 25px;
  box-shadow: 0 0 5px darkslategray;
`;