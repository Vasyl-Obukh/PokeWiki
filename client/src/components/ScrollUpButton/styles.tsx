import styled from 'styled-components';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

export const Button = styled(FontAwesomeIcon)`
  position: fixed;
  display: block;
  bottom: 150px;
  right: calc((100vw - ${props => props.theme.basicWidth}) / 2);
  height: 50px;
  padding: 10px;
  border-radius: 50%;
  background-color: ${props => props.theme.backgroundPrimary};
  font-size: 25px;
  color: white;
  cursor: pointer;

  && {
    width: 50px;
  }
`;