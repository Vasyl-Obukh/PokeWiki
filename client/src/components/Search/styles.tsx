import styled from 'styled-components';
import {Field} from 'redux-form';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

export const Form = styled.form`
  position: relative;
`;

export const StyledField = styled(Field)`
  padding: 10px 15px 10px 35px;
  width: 150px;
  border-radius: 25px;
  border: none;
  outline: none;
  background-color: #fff;
  color: black;
  box-shadow: 0 2px 5px darkslategray;
  transition: all .5s;
  
  &:focus {
    width: 350px;
    box-shadow: -2px 2px 5px darkslategray;
  }
`;

export const SearchIcon = styled(FontAwesomeIcon)`
  position: absolute;
  left: 10px;
  top: calc(50% - 8px);
`;