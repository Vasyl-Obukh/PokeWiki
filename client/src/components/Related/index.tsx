import * as React from 'react';
import * as Styles from './styles';
import getPokemonPageUrl from '../../utils/pageUrl';
import { BasicPokemonShape } from '../../global_interfaces/pokemon';

type Props = {
  data: BasicPokemonShape[]
}

const Related = (props: Props) => {
  const { data = [] } = props;

  return (
      <>
        <Styles.Title>Related pokemons</Styles.Title>
        <Styles.Wrapper>
          {data.map((_) => (
            <Styles.Item key={_.id} href={getPokemonPageUrl(_.id)}>
              <Styles.Image src={_.thumb} alt={_.name} />
              <Styles.Name>{_.name}</Styles.Name>
            </Styles.Item>
          ))}
        </Styles.Wrapper>
      </>
  );
};

export default Related;