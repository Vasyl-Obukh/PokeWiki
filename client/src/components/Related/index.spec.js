import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Related from './index';
import * as Styles from './styles';

Enzyme.configure({adapter: new Adapter()});

describe('<Related /> tests', () => {
  let props;

  beforeEach(() => {
    props = {
      data: [
        {id: 14, name: 'bulbasaur', thumb: '/some/url/1'},
        {id: 2, name: 'pikachu', thumb: '/some/url/2'}
      ]
    };
  });

  test('Should render correctly with no missing props', () => {
    const component = renderer.create(<Related {...props}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly if non-required props is missing', () => {
    delete props.data[0].thumb;

    const component = renderer.create(<Related {...props}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly if data prop is missing', () => {
    delete props.data;

    const component = renderer.create(<Related {...props}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Id prop should be contained in href attr of related card', () => {
    delete props.data[1];
    const component = mount(<Related {...props}/>);
    expect(component.find(Styles.Item).prop('href')).toContain(props.data[0].id);
  });

  test('Thumb prop should be equal to src attr of img tag', () => {
    delete props.data[1];
    const component = mount(<Related {...props}/>);
    expect(component.find(Styles.Image).prop('src')).toBe(props.data[0].thumb);
  });

  test('Name prop should be equal to text value of proper node', () => {
    delete props.data[1];
    const component = mount(<Related {...props}/>);
    expect(component.find(Styles.Name).text()).toBe(props.data[0].name);
  });
});