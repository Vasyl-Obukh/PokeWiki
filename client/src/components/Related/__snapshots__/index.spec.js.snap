// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Related /> tests Should render correctly if data prop is missing 1`] = `
Array [
  <h3
    className="sc-bwzfXH gJIwTT"
  >
    Related pokemons
  </h3>,
  <div
    className="sc-bdVaJa duObhc"
  />,
]
`;

exports[`<Related /> tests Should render correctly if non-required props is missing 1`] = `
Array [
  <h3
    className="sc-bwzfXH gJIwTT"
  >
    Related pokemons
  </h3>,
  <div
    className="sc-bdVaJa duObhc"
  >
    <a
      className="sc-htpNat dJebTn"
      href="/pokemon/14"
    >
      <img
        alt="bulbasaur"
        className="sc-bxivhb VdiwF"
      />
      <p
        className="sc-ifAKCX YYcQB"
      >
        bulbasaur
      </p>
    </a>
    <a
      className="sc-htpNat dJebTn"
      href="/pokemon/2"
    >
      <img
        alt="pikachu"
        className="sc-bxivhb VdiwF"
        src="/some/url/2"
      />
      <p
        className="sc-ifAKCX YYcQB"
      >
        pikachu
      </p>
    </a>
  </div>,
]
`;

exports[`<Related /> tests Should render correctly with no missing props 1`] = `
Array [
  <h3
    className="sc-bwzfXH gJIwTT"
  >
    Related pokemons
  </h3>,
  <div
    className="sc-bdVaJa duObhc"
  >
    <a
      className="sc-htpNat dJebTn"
      href="/pokemon/14"
    >
      <img
        alt="bulbasaur"
        className="sc-bxivhb VdiwF"
        src="/some/url/1"
      />
      <p
        className="sc-ifAKCX YYcQB"
      >
        bulbasaur
      </p>
    </a>
    <a
      className="sc-htpNat dJebTn"
      href="/pokemon/2"
    >
      <img
        alt="pikachu"
        className="sc-bxivhb VdiwF"
        src="/some/url/2"
      />
      <p
        className="sc-ifAKCX YYcQB"
      >
        pikachu
      </p>
    </a>
  </div>,
]
`;
